stories/core/Table.stories.tsx

```diff
@@ -307,6 +308,98 @@ Sortable.args = {
   isSortable: true,
 };
 
+export const Expandable: Story<TableProps> = (args) => {
+  const { columns, data, ...rest } = args;
+
+  const onExpand = useCallback(
+    (state) => action(`Expanded. Table state: ${JSON.stringify(state)}`)(),
+    [],
+  );
+
+  const tableColumns = useMemo(
+    () => [
+      {
+        Header: 'Table',
+        columns: [
+          {
+            id: 'expander',
+            Cell: (props: CellProps<{ name: string; description: string }>) => (
+              <IconButton
+                styleType='borderless'
+                size='small'
+                onClick={() => {
+                  props.row.toggleRowExpanded();
+                }}
+              >
+                {props.row.isExpanded ? (
+                  <SvgChevronDown />
+                ) : (
+                  <SvgChevronRight />
+                )}
+              </IconButton>
+            ),
+            width: 60,
+          },
+          {
+            id: 'name',
+            Header: 'Name',
+            accessor: 'name',
+          },
+          {
+            id: 'description',
+            Header: 'Description',
+            accessor: 'description',
+            maxWidth: 200,
+          },
+        ],
+      },
+    ],
+    [],
+  );
+
+  const tableData = useMemo(
+    () => [
+      { name: 'Name1', description: 'Description1' },
+      { name: 'Name3', description: 'Description3' },
+      { name: 'Name2', description: 'Description2' },
+    ],
+    [],
+  );
+
+  const expandedSubComponent = useCallback(
+    (row) => (
+      <div style={{ padding: 16 }}>
+        <Leading>Extra information</Leading>
+        <pre>
+          <code>{JSON.stringify({ values: row.values }, null, 2)}</code>
+        </pre>
+      </div>
+    ),
+    [],
+  );
+
+  return (
+    <Table
+      columns={columns || tableColumns}
+      data={data || tableData}
+      emptyTableContent='No data.'
+      expandedSubComponent={expandedSubComponent}
+      onExpand={(newState) => {
+        onExpand(newState);
+      }}
+      {...rest}
+    />
+  );
+};
+
+Expandable.args = {
+  data: [
+    { name: 'Name1', description: 'Description1' },
+    { name: 'Name3', description: 'Description3' },
+    { name: 'Name2', description: 'Description2' },
+  ],
+};
+
```

Comment:
Also, I think we will need a story with full features on.
THis way we could test more real life scenarios and also see, how features work together.
For this case, we have a standard, how selected expanded table row looks like, so. need to check that:
![image](https://user-images.githubusercontent.com/81580355/117618373-845ee180-b176-11eb-85e6-4d7db0f574b0.png)

Not necessarily for this PR, we could create a separate one.
@bentleyvk @mayank99

Reply:
![image](https://user-images.githubusercontent.com/36437897/117657404-2e079800-b1a2-11eb-91a5-b7581e154593.png)
Added another storybook entry for that.

Reply:
After further experimenting with this, I noticed a couple of issues when combining multiple features. For instance, upon sorting, bottom row reached gets triggered which results in new data being generated, after that sorting simply gets reset. I think this needs to go on a separate PR where fixes could be applied as well.

Reply:
Lets leave this full-on story for separate PR and fixes gonna be there. In this just have a story for expander.